File: ruby/hamming/hamming.rb
module Hamming
    def self.compute string1, string2
        raise ArgumentError.new unless string1.length == string2.length
        distance = 0
        (0..string1.length).each{|index| distance += string1[index] == string2[index] ? 0 : 1 }
        distance
    end
end

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
	if(#a ~= #b) then return -1 end
	local distance = 0
	for i = 1, #a do
		distance = distance + ( a:sub(i,i) == b:sub(i,i) and 0 or 1 )
	end
	return distance
end

return Hamming
